project/src/main/world/environment/shadow-caster.gd

```diff
@@ -0,0 +1,13 @@
+extends KinematicBody2D
+
+export (float) var shadow_scale: float = 1.0 setget set_shadow_scale
+
+func _init() -> void:
+	set_meta("shadow_scale", shadow_scale)
+	set_meta("chat_path", "res://assets/main/dialog/marsh-crystal.json")
```

Comment:
The chat path is specific to marsh-crystals; not geenric for all shadow casters.

Reply:
`chat_path` is now an exported property, this class is generic.